### PR TITLE
Add logging for data directory and config file paths

### PR DESCRIPTION
**Pull request by anipaul2**
Following up from #210 

fixes: #202

### CODE REVIEW

**Comment by sr-gi** [comment]
You really didn't need to open another PR for this. You should just have to rebase the old one. Please do not do this, it makes it hard to follow the review process. It's the second time already that this happens for the same issue.

**Comment by anipaul2** [comment]
> You really didn't need to open another PR for this. You should just have to rebase the old one. Please do not do this, it makes it hard to follow the review process. It's the second time already that this happens for the same issue.

I actually didn't closed it by myself. It happened when I synced on my fork and it got deleted. I apologize for creating a new PR and making it difficult to follow the review process. I understand the importance of keeping the conversation and history in a single PR. In the future, I will make sure to rebase and fix any issues within the existing PR. Thank you for your feedback, and I will take it into account for future contributions.

**Comment by anipaul2** [comment]
Fork repo says:-

image

But I have only pushed one commit and the other one gets attached automatically. I git logged and go through it but didn't found the second commit hash there so that I can reset. When I try to do rebase it shows my first commit only and second commit is not there. If it is there I can either drop or squash but it is not even present there. 

image

How can I solve this issue?

**Comment by anipaul2** [comment]
This is exactly what happened  with my previous pr so I synced from my repo itself which resulted in pr closed. I never did intentionally. I don't know how to solve this. @sr-gi please if you could help regarding this it would really help me understand how this works as a whole.

**Comment by sr-gi** [comment]
You needed to first pull from upstream to get the new changes. Normally you'd do this on master, but you were already working on master instead of in a separate branch for this PR. After having your local master synced with upstream, you would need to rebase your branch with master, which may cause some merge issues that you'd need to deal with. After that you can force push.

**Comment by anipaul2** [comment]
> You needed to first pull from upstream to get the new changes. Normally you'd do this on master, but you were already working on master instead of in a separate branch for this PR. After having your local master synced with upstream, you would need to rebase your branch with master, which may cause some merge issues that you'd need to deal with. After that you can force push.

I did all of these but didn't work. 

image

The second one here is not even committed be me. When I click on this commit it shows these:-

image

The extra commit hash is d698a26751ed39a56c6b859ad603c1bd0592bc82 which has two parents one from previous pr that got deleted and the other one is fix clippy responder one.

2 parents [4bde9ff](https://github.com/talaia-labs/rust-teos/commit/4bde9ffde2ad7d8caf97fb301039d3812d950a8d) + [886e0ff](https://github.com/talaia-labs/rust-teos/commit/886e0fffe5cb0e6f0e87edfefd60f4a322a8c6ee)
commit d698a26

I am not able to solve this and can do by making a new branch related to this issue and the cherry pick the first commit that I want to keep and then force push.

**Comment by anipaul2** [comment]
When I remove the second commit (latest one) and point the head to the first one it says branch is outdated and fork becomes one commit ahead which I did and one commit behind which was from @sr-gi. When I sync it to match the upstream then the fork gets updated with two commits and shows two commits ahead. @mariocynicys @sr-gi I understand that you both are quite busy, and I truly appreciate any time and assistance you can provide to help me understand and resolve this issue.

**Comment by mariocynicys** [comment]
I'm not really following on the git issues you are facing @anipaul2, but I think you wanna give [git book](https://git-scm.com/book/en/v2) a read first as it gonna help your hands get moving with git. We don't have to rush on this PR for now.

**Comment by sr-gi** [comment]
I'm not following either at this point 🫤

**Comment by anipaul2** [comment]
> I'm not following either at this point 🫤

In summary I added and committed before fetching the upstream at first so that's why it became so confusing. And it cannot be resolved too in this branch. I have to raise another pr with another branch and not witih master then it will work. But, I guess I already raise two so if you agree I will do.

**Comment by sr-gi** [comment]
I'll fix it

**Comment by anipaul2** [comment]
> I'll fix it

I sincerely apologize for the inconvenience caused by the issues with my PR. I understand that it has made the review process more challenging, and I want to assure you that I will take the necessary steps to avoid this situation in the future.

**Comment by sr-gi** [comment]
There you go, your master is now up to date with upstream.

Pull that so your local master looks the same, then apply your commit on top (don´t worry if you delete your data, I saved your commit here: https://github.com/sr-gi/rust-teos/commit/16ff91aa51985d60eb29d59d8a3cf285dd5d344b).

Once you push your master with your new commit to your remote, this PR should reopen automatically.

**Comment by anipaul2** [comment]
> There you go, your master is now up to date with upstream.
> 
> Pull that so your local master looks the same, then apply your commit on top (don´t worry if you delete your data, I saved your commit here: [sr-gi@16ff91a](https://github.com/sr-gi/rust-teos/commit/16ff91aa51985d60eb29d59d8a3cf285dd5d344b)).
> 
> Once you push your master with your new commit to your remote, this PR should reopen automatically.

Should I delete the two local commits?

image

and then write the code again from the saved data as you said or cherry pick the commit and push it?

**Comment by anipaul2** [comment]
I did git reset hard and local commits removed. Adding my codes from the data saved and will raise a pr soon.

**Comment by anipaul2** [comment]
> Following up from #210
> 
> fixes: #202

Thank you so much for the saved data! I pushed the commit but it doesn't reopens this pr and shows conflicts again so added manually and raised a pr again. I know raising pr three times for this simple issue makes the process of review very difficult that's why I tried my best but doing further research I got that my local commits have messed up totally and I have to raise a pr again. I am really sorry for the inconvenience caused and from next time I will not make the same mistake again.
